### PR TITLE
add spelling checker

### PR DESCRIPTION
**Pull request by hormes**
Signed-off-by: Fansong Zeng <fanster.z@gmail.com>

issue: #29

### CODE REVIEW

**Comment by jasonliu747** [comment]
fix #29

**Comment by jasonliu747** [comment]
Fix #29

File: .github/workflows/spelling_action.yml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+name: Spellcheck Action
+on: push
```
Comment (FillZpp) [inline .github/workflows/spelling_action.yml:2]:
```yaml
on:
  push: {}
  pull_request: {}
```

Comment (hormes) [reply .github/workflows/spelling_action.yml]: done

Comment (FillZpp) [reply .github/workflows/spelling_action.yml]: now you can see the failed check...

**Review by jasonliu747 — CHANGES_REQUESTED** [review]
Need a rich word list.

**Review by jasonliu747 — APPROVED** [review]
/lgtm

**Comment by hormes** [comment]
word list will be too long to maintain

**Comment by jasonliu747** [comment]
there are severals actions provided by reviewdog, we can use them to do spell check.
https://github.com/reviewdog/reviewdog

**Pull request closed without merging**